src/calling/haplotype_abundances.rs

```diff
@@ -1,43 +1,38 @@
+use anyhow::Result;
+use derive_builder::Builder;
 use hdf5;
+use kernel_density;
 use rust_htslib::bcf;
-use derive_builder::Builder;
-use anyhow::Result;
-use std::path::Path;
+
 
 #[derive(Builder)]
 #[builder(pattern = "owned")]
-pub(crate) struct Caller<'a> {
-    hdf5_reader: hdf5::Reader<'a>,
+pub(crate) struct Caller {
+    hdf5_reader: hdf5::File,
     vcf_reader: bcf::Reader,
 }
 
-impl<'a> Caller<'a> {
+impl Caller {
+    pub(crate) fn cdf(&self) -> Result<()> {
```

Comment:
Good! I would expect this to have a parameter for the sequence name though. With regular kallisto usage, it would be the transcript id, in our case it would be the name of the respective contig (e.g. the HLA type, or the virus strain).

We will call this function for each of them.

Reply:
I did some changes, could you have a look at it again?